src/main/java/io/cdap/plugin/aws/s3/sink/S3BatchSink.java

```diff
@@ -194,6 +196,15 @@ public String getPath() {
       return path;
     }
 
+    @Override
+    public String getPath(@Nullable StageContext context) {
+      if (!path.startsWith(S3Path.OLD_SCHEME)) {
+        return path;
+      }
+      boolean useS3n = context == null ? false : Boolean.valueOf(context.getArguments().get(S3Constants.USE_S3N));
```

Comment:
similar question in the other PR, but instead of having context be nullable, seems better to handle that case in getPath()? Otherwise the behavior here when context is null and the behavior of getPath() can be inconsistent.

Reply:
I thought the method will be called in configurePipeline, but it is actually only called in prepareRun, so context should never be null, removed nullable here